app/controllers/stories_controller.rb

```diff
@@ -127,6 +127,27 @@ def move
     redirect_to @project
   end
 
+  def approve
+    @story.approved!
+    respond_to do |format|
+      format.js { render "shared/update_status" }
+    end
+  end
+
+  def reject
+    @story.rejected!
+    respond_to do |format|
+      format.js { render "shared/update_status" }
+    end
+  end
+
+  def pending
+    @story.pending!
+    respond_to do |format|
+      format.js { render "shared/update_status" }
+    end
+  end
+
```

Comment:
This looks weird to me because we will need to create a new endpoint if in the future we decide to add a new story's status.

what about doing something like so:

```ruby

def update_status
  @story.update(status: params[:status]) # pass a number
  # or
  @story.try(params[:status]) # where the param is a string like `pending!`, `rejected!`, etc. 
    
  respond_to do |format|
    format.js { render "shared/update_status" }
  end
end
```

```ruby
@story.update(status: params[:status]) 
irb(main):003:0> story.update(status: 0)
  TRANSACTION (0.1ms)  BEGIN
  Project Load (0.2ms)  SELECT "projects".* FROM "projects" WHERE "projects"."id" = $1 LIMIT $2  [["id", 4], ["LIMIT", 1]]
  TRANSACTION (0.2ms)  COMMIT
=> true
irb(main):004:0> story.pending?
=> true
irb(main):005:0> story.rejected?
=> false
```

```ruby
@story.try(params[:status]) 
irb(main):006:0> story.try("approved!")
  TRANSACTION (0.1ms)  BEGIN
  Story Update (0.3ms)  UPDATE "stories" SET "updated_at" = $1, "status" = $2 WHERE "stories"."id" = $3  [["updated_at", "2023-10-09 20:38:42.837573"], ["status", 1], ["id", 8]]
  TRANSACTION (0.4ms)  COMMIT
=> true
irb(main):007:0> story.pending?
=> false
irb(main):008:0> story.approved?
=> true
```

Reply:
this was the original and I asked to replace that with individual actions per status lol

I don't think it hurts to have new endpoints, it improves the readability (it's more clear to read `@story.approve!` than `@story.update(status: params[:status])`)

Reply:
Okay, that sounds good to me.